### PR TITLE
[WIP] series_list plugin 

### PR DESCRIPTION
**Pull request by stevezau**
### Motivation for changes:

Add an internal way to hold a series list

### Detailed changes:
- Added managed list series_list plugin
- Created SeriesList DB

### Addressed issues:

- Fixes # .

### Config usage if relevant (new plugin or updated schema):
```
  test_series_list:
    mock:
      - {series_name: "lucifer", title: "lucifer"}
    list_add:
      - series_list: series_list
    accept_all: yes
```
#### To Do:
- [ ] Add tests for configure_series
- [ ] Look at CLI

### CODE REVIEW

File: flexget/plugins/cli/series.py [hunk 1]
```diff
@@ -39,22 +49,30 @@ def do_cli(manager, options):
...
 
 
 def display_summary(options):
     """
     Display series summary.
     :param options: argparse options from the CLI
     """
-    porcelain = options.table_type == 'porcelain'
     with Session() as session:
+        list = get_list(options.list_name, session)
```
Comment (liiight) [inline flexget/plugins/cli/series.py]: Overriding builtin

File: flexget/plugins/cli/series.py [hunk 2]
```diff
@@ -112,27 +130,40 @@ def display_summary(options):
...
-        except ValueError as e:
-            console(e)
-        else:
-            console('Episodes for `%s` will be accepted starting with `%s`' % (series.name, ep_id))
-            session.commit()
+        if ep_id:
+            try:
+                set_series_begin(series, ep_id)
+            except ValueError as e:
+                console(e)
```
Comment (liiight) [inline flexget/plugins/cli/series.py]: Should probably be `str(e)`

File: flexget/plugins/list/series_list.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            return
+        log.debug('trying to match series based of name: %s and year: %s', name, year)
+        res = (self._db_list(session).series.filter(func.lower(SeriesListSeries.title) == name.lower())
+               .filter(SeriesListSeries.year == year).first())
+        if res:
+            log.debug('found series %s', res)
+        return res
+
+    @staticmethod
+    def _parse_title(entry):
```
Comment (liiight) [inline flexget/plugins/list/series_list.py]: Did you use this somewhere?

Comment (stevezau) [inline flexget/plugins/list/series_list.py]: nah, copied from movie, ill delete it

File: flexget/plugins/list/series_list.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+
+class SeriesListBase(object):
+    """
+    Class that contains helper methods for series list as well as plugins that use it,
+    such as API and CLI.
+    """
+
+    @property
+    def supported_ids(self):
```
Comment (gazpachoking) [inline flexget/plugins/list/series_list.py]: Can't this just be a function? Why make put it inside this class?

File: flexget/plugins/list/series_list.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    def __repr__(self):
+        return '<SeriesListSeries title=%s,year=%s,list_id=%d>' % (self.title, self.year, self.list_id)
+
+    def to_entry(self, strip_year=False):
+        entry = Entry()
+        entry['title'] = entry['series_name'] = self.title
+        entry['url'] = 'mock://localhost/series_list/%d' % self.id
+        entry['added'] = self.added
+        if self.year:
+            if strip_year is False:
```
Comment (gazpachoking) [inline flexget/plugins/list/series_list.py]: If strip_year is false, it adds the year to the title?

File: flexget/plugins/list/series_list.py [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    ids = relationship('SeriesListID', backref='series', cascade='all, delete, delete-orphan')
+
+    def __repr__(self):
+        return '<SeriesListSeries title=%s,year=%s,list_id=%d>' % (self.title, self.year, self.list_id)
+
+    def to_entry(self, strip_year=False):
+        entry = Entry()
+        entry['title'] = entry['series_name'] = self.title
+        entry['url'] = 'mock://localhost/series_list/%d' % self.id
+        entry['added'] = self.added
```
Comment (gazpachoking) [inline flexget/plugins/list/series_list.py]: Should this also populate series_name?

File: flexget/plugins/list/series_list.py [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+        return False
+
+    @with_session
+    def get(self, entry, session):
+        match = self._find_entry(entry=entry, session=session)
+        return match.to_entry() if match else None
+
+
+class PluginSeriesList(object):
+    """Remove all accepted elements from your trakt.tv watchlist/library/seen or custom list."""
```
Comment (gazpachoking) [inline flexget/plugins/list/series_list.py]: Remove what now?

File: flexget/plugins/cli/series.py [hunk 8]
```diff
@@ -39,22 +49,30 @@ def do_cli(manager, options):
         remove(manager, options)
     elif options.series_action == 'forget':
         remove(manager, options, forget=True)
-    elif options.series_action == 'begin':
-        begin(manager, options)
```
Comment (gazpachoking) [inline flexget/plugins/cli/series.py]: What happened to 'begin'?

File: flexget/plugins/cli/series.py [hunk 9]
```diff
@@ -39,22 +49,30 @@ def do_cli(manager, options):
...
     Display series summary.
     :param options: argparse options from the CLI
     """
-    porcelain = options.table_type == 'porcelain'
     with Session() as session:
+        list = get_list(options.list_name, session)
+        if not list:
+            console('Could not find list with name: `%s`' % options.list_name)
+            return
+        list_id = list.id
```
Comment (gazpachoking) [inline flexget/plugins/cli/series.py]: Is there some reason this isn't silly? Because it looks silly.

File: flexget/plugins/cli/series.py [hunk 10]
```diff
@@ -112,27 +130,40 @@ def display_summary(options):
         console(footer)
 
 
-def begin(manager, options):
+def manage(manager, options, update=False):
```
Comment (gazpachoking) [inline flexget/plugins/cli/series.py]: Wait, what is even going on here? Why is the series plugin cli even being modified? I thought this series_list implementation was separate from series plugin?

Comment (liiight) [inline flexget/plugins/cli/series.py]: This is a leftover from the 2nd pr imo...

**Pull request closed without merging**